pkg/policy/distillery/policy.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package distillery
+
+import (
+	"sync/atomic"
+	"unsafe"
+
+	identityPkg "github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/identity/cache"
+	"github.com/cilium/cilium/pkg/policy"
+)
+
+// cachedSelectorPolicy is a wrapper around a SelectorPolicy (stored in the
+// 'policy' field). It is always nested directly in the owning policyCache,
+// and is protected against concurrent writes via the policyCache mutex.
+//
+// 'policy' and 'revesion' are only consistent if queried while holding a lock.
```

Comment:
->'revision', but then the 'revision' is no longer here, so needs an update?

Reply:
I'll tidy it up in a follow-up PR.